core/src/main/java/bisq/core/dao/state/model/DaoState.java

```diff
@@ -98,6 +102,11 @@ public static DaoState getClone(DaoState daoState) {
     @Getter
     private final List<DecryptedBallotsWithMerits> decryptedBallotsWithMeritsList;
 
+    // Transient data used only as an index - must be kept in sync with the block list
+    @Getter
+    @JsonExclude
+    private transient final Map<String, Tx> txMap; // key is txId
```

Comment:
It's using a hash table rather than a tree set, so I don't think the additional memory will be a problem - there are only about 10,000 txs or so right now, so I don't think it will take up more than 100KB or so. (The fact that DaoState.blocks is a linked list instead of an array list is probably a more significant memory issue that could be easily fixed.)